fe/fe-core/src/main/java/com/starrocks/sql/optimizer/OutputPropertyDeriver.java

```diff
@@ -130,6 +119,15 @@ public PhysicalPropertySet computeHashJoinDistributionPropertyInfo(PhysicalHashJ
             propertyInfo.nullableColumns.union(leftChildColumns);
             propertyInfo.nullableColumns.union(rightChildColumns);
         }
+
+        if (node.getJoinType().isInnerJoin()) {
```

Comment:
I think we should consider the the output property of bucket shuffle join, for example :
![image](https://user-images.githubusercontent.com/9495145/163523353-c50962ee-fbf1-41c1-bc1f-f6a9e29ed0a3.png)
table B and C in the same colocate group, the will result in colocate join at the top join, but table B data has bucket shuffle to Table A, not satisfythe colocate requirement, this  will cause error result

Reply:
Emm, the top join will be bucket shuffle join but not colocate join

The case you memtioned have been added to ut, as follows:

* `colocate_t0` and `colocate_t1` are in one colocate group
* `colocate_t2` and `colocate_t3` are in another colocate group

```java
            String sql =
                    "select * from (select * from colocate_t0 join[bucket] colocate_t2 on colocate_t0.v1 = colocate_t2.v7) s1 " +
                            "join [colocate] colocate_t3 on s1.v7 = colocate_t3.v10";
            String plan = getFragmentPlan(sql);
            assertContains(plan, "  |  join op: INNER JOIN (BUCKET_SHUFFLE)\n" +
                    "  |  hash predicates:\n" +
                    "  |  colocate: false, reason: \n" +
                    "  |  equal join conjunct: 4: v7 = 7: v10\n" +
                    "  |  \n" +
                    "  |----5:EXCHANGE\n" +
                    "  |    \n" +
                    "  3:HASH JOIN\n" +
                    "  |  join op: INNER JOIN (BUCKET_SHUFFLE)\n" +
                    "  |  hash predicates:\n" +
                    "  |  colocate: false, reason: \n" +
                    "  |  equal join conjunct: 1: v1 = 4: v7");
```